wagtail/admin/views/pages/listing.py

```diff
@@ -265,6 +265,14 @@ def get_table_kwargs(self):
         kwargs["actions_next_url"] = self.index_url
         return kwargs
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        if any(isinstance(column, ParentPageColumn) for column in self.columns):
+            Page.objects.annotate_parent_page(context["object_list"])
```

Comment:
Random musing (definitely out of scope for this PR) - I wonder if this calls for columns to have a mechanism for playing a part in building the queryset, so that an annotation is brought in automatically whenever you use a particular column. Of course, at that point wouldn't really be "UI components" any more :-)

Reply:
Yep, I wondered the same thing. The `UpdatedAtColumn` would be able to make use of this mechanism as well, so developers don't have to worry about adding the annotation.

The tricky part I think is that depending on the optimisation, the annotation can be done in two places: during the queryset building with Django's QuerySet APIs vs. after pagination has been applied. If we add this mechanism, it has to support both cases.